public/app/plugins/panel/heatmap-new/fields.ts

```diff
@@ -59,6 +60,9 @@ export function prepareHeatmapData(
   // detect a frame-per-bucket heatmap frame
   // TODO: improve heuristic? infer from fields[1].labels.le === '+Inf' ?
   if (frames[0].meta?.custom?.resultType === 'matrix' && frames.some((f) => f.name?.startsWith('+Inf'))) {
+    // already done by the Prometheus datasource frontend
+    //frames = prepBucketFrames(frames);
```

Comment:
this was decoupled and commented out since we currently assume pre-prepped frames arrive here.